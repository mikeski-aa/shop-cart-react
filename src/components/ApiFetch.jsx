// this function will be responsible for fetching cart items and images from the free shop api provided
// this should be stored in an object so that we don't have to call the API every time shop is clicked
// i guess caching it?

import { useEffect } from "react";

async function callShopApi() {
  try {
    const response = await fetch("https://fakestoreapi.com/products");

    if (!response.ok) {
      throw new Error(response.status);
    } else {
      const jsonFile = await response.json();
      return jsonFile;
    }
  } catch (error) {
    alert("Oops something went wrong!");
    return error;
  }
}

function getCallResults(setstate) {
  callShopApi().then((result) => {
    useEffect(() => {
      setstate(result);
    }, [result]);
  });
}

export { callShopApi, getCallResults };
