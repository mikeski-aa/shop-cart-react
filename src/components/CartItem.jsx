import "../styles/CartItem.css";

// function for rendering individual card items for the cart part of the store

function CartItem(props) {
  return (
    <div className="cartItem">
      <div className="cartImageDiv">
        <img className="cartImage" src={props.url}></img>
      </div>
      <div className="itemName">{props.name}</div>
      <div className="itemSubtotal">EUR 0</div>
      <div className="quantityChange">
        <button>-</button>
        <input type="number" value={props.quantity}></input>
        <button>+</button>
      </div>
    </div>
  );
}

export { CartItem };
