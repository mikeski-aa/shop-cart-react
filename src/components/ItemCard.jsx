import { ItemContext } from "../Shop";
import { useContext } from "react";
import "../styles/ItemCard.css";

// function for rendering the individual item card

function ItemCard(props) {
  const handleAddToCart = () => {
    console.log(props.id);
    return props.id;
  };

  return (
    <div className="itemCard">
      <div className="imageContainer">
        <img className="itemImage" src={props.image}></img>
      </div>
      <div className="itemNameAndPrice">
        <h3 className="itemName">{props.title}</h3>
        <h3 className="itemPrice">â‚¬ {props.price}</h3>
      </div>
      <div className="itemDescription">{props.description}</div>
      <div className="itemRating">{props.rating}</div>
      <button onClick={handleAddToCart}>Add to cart</button>
    </div>
  );
}

export { ItemCard };
