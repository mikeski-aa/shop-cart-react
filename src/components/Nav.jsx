import { Link } from "react-router-dom";
import "../styles/Nav.css";
// this function will create the navigation bar for the app
// this nav bar will be displayed on main page and shop page

function Nav() {
  return (
    <div className="navBar">
      <Link to="/">
        <button className="homeBtn">Homepage</button>
      </Link>

      <Link to="/shop">
        <button className="shopBtn">Shop</button>
      </Link>
    </div>
  );
}

export { Nav };
