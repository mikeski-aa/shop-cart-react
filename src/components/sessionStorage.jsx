// function here will handle all sessionStorage operations
// using sessionStorage over localstorage as I do not want shop data to save between browser close,
// but instead only when switching between shop and main page urls

function setSessionStorageData(key, dataToStore) {
  console.log("data stored");
  console.log(dataToStore);
  let toString = JSON.stringify(dataToStore);
  sessionStorage.setItem(key, toString);
  console.log(toString);
}

function getSessionStorageData(key) {
  let x = sessionStorage.getItem(key);
  let retrieved = JSON.parse(x);
  console.log("data retrieved");
  console.log(retrieved);
  return retrieved;
}

export { setSessionStorageData, getSessionStorageData };
