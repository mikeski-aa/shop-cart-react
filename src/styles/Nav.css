.navBar {
  display: flex;
  flex-direction: column;
}

.containerRandom {
  width: 200px;
}
