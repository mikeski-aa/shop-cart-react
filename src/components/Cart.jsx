import { useContext, useEffect, useState } from "react";
import { ItemContext } from "../Shop";
import { CartItem } from "./CartItem";

// function for getting the total value of the cart
function getTotalCartValue(itemContext) {
  let newTotal = 0;
  let newSubtotal = 0;

  for (let x of itemContext.cart) {
    newSubtotal = x.itemQuantity * x.price;
    newTotal += newSubtotal;
  }

  return Math.round(newTotal * 100) / 100;
}

// function for displaying current cart
// in currentItems we will map items added to the cart

function Cart() {
  const itemContext = useContext(ItemContext);

  if (itemContext.storePage != 6) {
    return null;
  }

  useEffect(() => {}, [itemContext.cart]);

  return (
    <div className="currentCart">
      <div className="currentItems">
        <h2>Your basket</h2>
        {itemContext.cart.map((item) => (
          <CartItem
            key={item.id}
            url={item.image}
            name={item.title}
            quantity={item.itemQuantity}
            subtotal={+item.itemQuantity * +item.price}
            id={item.id}
          />
        ))}
      </div>
      <div className="totalOrder">
        <div className="containerOrderTotal">
          <h2>Order total</h2>
          <div className="totalOrderValue">
            â‚¬ {getTotalCartValue(itemContext)}
          </div>
          <button className="checkoutBtn">Proceed to checkout</button>
        </div>
      </div>
    </div>
  );
}

export { Cart };
