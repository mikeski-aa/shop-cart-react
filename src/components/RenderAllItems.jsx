import { useContext } from "react";
import { ItemContext } from "../Shop";
import { ItemCard } from "./ItemCard";

// function to render shop cards depending on input
function RenderAllItems() {
  const itemContext = useContext(ItemContext);

  if (itemContext.storePage != 1) {
    return null;
  }

  return (
    <div className="cards">
      {itemContext.items.map((item) => (
        <ItemCard
          key={item.id}
          image={item.image}
          title={item.title}
          price={item.price}
          description={item.description}
          rating={item.rating.rate}
          id={item.id}
        />
      ))}
    </div>
  );
}

export { RenderAllItems };
