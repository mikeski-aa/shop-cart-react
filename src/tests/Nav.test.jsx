import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { describe, expect, it } from "vitest";
import { Nav } from "../components/Nav";

describe("testing navigation bar", () => {
  it("check snapshot render", () => {
    const { container } = render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
