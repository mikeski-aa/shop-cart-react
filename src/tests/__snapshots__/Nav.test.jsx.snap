// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing navigation bar > check snapshot render 1`] = `
<div>
  <div
    class="navBar"
  >
    <button>
      <a
        href="/"
      >
        Homepage
      </a>
    </button>
    <button>
      <a
        href="/shop"
      >
        Shop
      </a>
    </button>
  </div>
</div>
`;
